src/Generator/JsonGenerator.php

```diff
@@ -37,28 +37,49 @@ public function __construct(Metas $metas, BuildConfig $buildConfig)
         $this->buildConfig = $buildConfig;
     }
 
-    public function generateJson()
+    /**
+     * Returns an array of each JSON file string, keyed by the input filename
+     *
+     * @param string $masterDocument The file whose toctree should be read first
+     * @return string[]
+     */
+    public function generateJson(string $masterDocument = 'index'): array
     {
         $fs = new Filesystem();
 
         $progressBar = new ProgressBar($this->output ?: new NullOutput());
         $progressBar->setMaxSteps(\count($this->metas->getAll()));
 
+        $walkedFiles = [];
+        $tocTreeHierarchy = $this->walkTocTreeAndReturnHierarchy(
+            $masterDocument,
+            $walkedFiles
+        );
+        // for purposes of prev/next/parents, the "master document"
+        // behaves as if it's the first item in the toctree
+        $tocTreeHierarchy = [$masterDocument => []] + $tocTreeHierarchy;
+        $flattenedTocTree = $this->flattenTocTree($tocTreeHierarchy);
+
+        $fJsonFiles = [];
         foreach ($this->metas->getAll() as $filename => $metaEntry) {
             $parserFilename = $filename;
             $jsonFilename = $this->buildConfig->getOutputDir().'/'.$filename.'.fjson';
 
             $crawler = new Crawler(file_get_contents($this->buildConfig->getOutputDir().'/'.$filename.'.html'));
 
+            $next = $this->determineNext($parserFilename, $flattenedTocTree, $masterDocument);
+            $prev = $this->determinePrev($parserFilename, $flattenedTocTree);
             $data = [
                 'title' => $metaEntry->getTitle(),
+                'parents' => $this->determineParents($parserFilename, $tocTreeHierarchy) ?: [],
                 'current_page_name' => $parserFilename,
                 'toc' => $this->generateToc($metaEntry, current($metaEntry->getTitles())[1]),
-                'next' => $this->guessNext($parserFilename),
-                'prev' => $this->guessPrev($parserFilename),
+                'next' => $next,
+                'prev' => $prev,
                 'rellinks' => [
```

Comment:
This is the official Sphinx explanation of `rellinks`: https://www.sphinx-doc.org/en/master/templating.html#rellinks

I'd leave this as it is because I think we don't use this option anywhere.

Reply:
As we already have changed the format of the file anyway, we could remove rellinks support as we don't use it anywhere AFAIK.

Reply:
I have removed it. If there is other information that would be useful, let me know and we an add it :)